### ISSUE TITLE
Create view of publiclab.org/tags that excludes powertags

### ISSUE THREAD

**Issue by ebarry** [publiclab/plots2#3482]
Hi everyone, 
I wanted to begin with how wonderful it is to use the page https://publiclab.org/tags. I frequently use this page to discover new tags, to make sure i'm adding relevant tags to connect content to other content, and to see how many people are subscribed to each tag in order to compare to how many people are subscribed to a mailing list about the same topic (mailing lists were what we used before we had this website). 

### Please describe the problem (or idea)

> What happened just before the problem occurred? Or what problem could this idea solve?

Here's what i am encountering: there are a lot of tags. Not an unreasonable amount, but when considering that every unique `lat:###` and `lng:####` show up as individual tags, it may soon become unreasonable. Read more about powertags here: https://publiclab.org/wiki/power-tags. Powertags include a `:` in the middle of the tag. 

> What did you expect to see that you didn't?

**I would like to see a view of the /tags page that excludes powertags.** This page could live at someplace like https://publiclab.org/tags?type=basic

### Please show us where to look

https://publiclab.org/tags

### What's your PublicLab.org username?

> This can help us diagnose the issue: 

liz

**Comment by jywarren** [comment]
Oh, that would be great!!!!

**Comment by LonghornRach** [comment]
n/m i just found the definition in the model

**Comment by jywarren** [comment]
👍🏽😀

**Comment by LonghornRach** [comment]
think i got in over my head and need to hand this one back :-/

**Comment by jywarren** [comment]
Hey, no problem! I've made a new one that you can work on if you're interested, over at #3744. If you'd like to, please leave a comment over there!

**Comment by jywarren** [comment]
This one will require changes at the tag controller, to check if `params[:type] == 'basic'`, in this `index` action:

https://github.com/publiclab/plots2/blob/dd18e979e619005c28254fd129852602e0279dd7/app/controllers/tag_controller.rb#L5-L16

I think it would get an extra condition like:

```ruby
.where('term_data.name NOT LIKE (?)', '%:%') # <= excluding ':' in tag names!
```

I **think** you can just add such a line just before the final `end` of the `index` method:

https://github.com/publiclab/plots2/blob/dd18e979e619005c28254fd129852602e0279dd7/app/controllers/tag_controller.rb#L69-L71

So:

```ruby
      @tags = Tag.where(tid: ids).sort_by { |p| ids.index(p.tid) }.paginate(page: params[:page], per_page: 24)
    end
    @tags.where('term_data.name NOT LIKE (?)', '%:%') if params[:type] == 'basic'
  end
```

However, I could be wrong and it's possible this wouldn't work. The way to test it would be to write a test like this one, and inspect the returned tags to ensure that no `:` characters are in the tag names. And, to run the test **without** the `basic` filter to ensure that normally there ARE some tags with `:` in them, otherwise it wouldn't be a good test!

We'd love help with this, whether @LonghornRach or someone else is interested! and we're happy to help out.

**Comment by LonghornRach** [comment]
I'll keep working on this but please anyone else feel free to claim it!

**Comment by oorjitchowdhary** [comment]
I think I can try this.. I'm a GCI candidate..

**Comment by jywarren** [comment]
yes please!

**Comment by JonathanXu1** [comment]
Is anyone still working on this? I've implemented @jywarren 's changes to see if it would work, but nothing really changed. You can see more info at #4232

**Comment by ebarry** [comment]
We are still quite actively interested in your contributions for creating a view of publiclab.org/tags that excludes powertags , and also to be used in the visualization of tags as i commented on your excellent pull request. Thanks again for your leadership on this!

**Comment by jywarren** [comment]
Hi, this is still open although we closed #4232 -- I think the filter would have to be added higher in the page than it was here: https://github.com/publiclab/plots2/pull/4232/files

because there are sorts and other things which prevent the `where()` clause from affecting the query. So unfortunately we may have to add that query before lines like this, which are no longer maintaining an ActiveRecord query chain: https://github.com/publiclab/plots2/pull/4232/files#diff-a8eb2425ebeb0814d010864bb014e815R69

I think that means we'd have to add it redundantly to each type of tag query, which I know is a bit annoying and inefficient. 

One way to ensure that this works is to add tests first, following Test-driven development. We could start with something like this:


```ruby
get :index, params: { basic: true }
assert_not assigns(:tags).include?('activity:coding') # some tag from our fixtures which would show up in first page of results 

get :index, params: { basic: false }
assert assigns(:tags).include?('activity:coding')
```

In a test like this one:

https://github.com/publiclab/plots2/blob/master/test/functional/tag_controller_test.rb#L138-L146

We might add a /positive/ test to that original test, to show that it does in fact show power tags when we don't set `basic` to `true`. Like:

```
assert assigns(:tags).include?('activity:coding') 
```

**Comment by jywarren** [comment]
I believe this is resolved now. Please re-open and clarify if not, thank you!!! 🎉
